content/blog/endpoint-deprecation/index.md

```diff
@@ -0,0 +1,115 @@
+---
+path: '/endpoint-deprecation'
+title: 'Can I deprecate this endpoint?'
+date: 2022-01-11T06:00:00.000+00:00
+featured_image: hero-image.png
+categories: ['Pixie Team Blogs', 'Kubernetes']
+authors: ['Hannah Troisi']
+emails: ['htroisi@pixielabs.ai']
+---
+
+Nothing lasts forever, including even the best designed APIs.
+
+Let’s imagine you are a developer who has taken over ownership of a Catalog microservice. You’ve been asked to deprecate the `/v1/catalog` endpoint in favor of the new `/v2/catalog` endpoint. How do you go about this?
+
+Whatever the reason for removal – a new version or a planned end-of-life – the first step in a _graceful_ API deprecation is to observe:
+
+- Is this endpoint used?
+- If so, who is calling it?
+
+## Is this endpoint used?
+
+Before you can deprecate the endpoint, you need to first check if the endpoint is actually being used.
+
+### Search the codebase
+
+For internal endpoints, a great way to start is to search the codebase for calls to the API. However, once you believe all calls have been removed, you will still want to use observability tooling to verify that all usage of the API has indeed stopped.
```

Comment:
Do you think it's valuable to cite explicit reasons for why searching in the codebase isn't enough? For example, an older version of a service is still running.